teps/0053-nested-triggers.md

```diff
@@ -201,4 +202,86 @@ spec:
 Notice how simple it is for the Trigger author to write the logic around their trigger
 when the common logic is implemented at the top level. Trigger authors can also manage
 their own set of downstream triggers as you can specify additional triggers alongside
-a binding and template once the interceptor chain completes.
\ No newline at end of file
+a binding and template once the interceptor chain completes.
+
+## Alternatives
+
+There are three alternatives that have been identified to provide some of the same benefits.
+
+### Option 1: Select Downstream Triggers With Labels
+
+Add a `triggerSelector` field that allows a trigger to target downstream triggers based off a label selector:
+
+```yaml
+apiVersion: triggers.tekton.dev/v1alpha1
+kind: Trigger
+metadata:
+  name: top-level-trigger
+spec:
+  interceptors:
+    - cel:
+        filter: >-
+          header.match('header-filter', 'a-header-filter')
+        - key: example-overlay
+          expression: body.example.value
+  triggerSelector:
+    matchLabels:
+      key: value
+```
+
+### Option 2: Interceptor Groups
+
+Specify the interceptor groups at the top level and process each group on trigger:
+
+```yaml
+apiVersion: triggers.tekton.dev/v1alpha1
+kind: EventListener
+metadata:
+  name: my-el
+spec:
+  groups: 
+  - name: github-group
+    interceptors:
+      - my-interceptor
+      - add-extra-body
+    triggerSelector:
+      matchLabels:
+        app=foo
+        anotherval=xyz
+  - name: another-group
+    interceptors:
+      - my-interceptor
+    triggerSelector:
+       matchLabels:
+         app=foo
+```
+
+This can work well with running a root EventListener interceptor with a `labelSelector`. One challenge here is that these `InterceptorGroups` become another separate API resource that would need to be managed and potentially added to the API. This could be added onto the existing Trigger resource, but then it becomes closer to Option 1.
+
+### Option 3: Downstream EventListener
+
+Allow triggers to target other eventListeners after processing interceptors:
+
+```yaml
+apiVersion: triggers.tekton.dev/v1alpha1
+kind: Trigger
+metadata:
+  name: top-level-trigger
+spec:
+  interceptors:
+    - cel:
+        filter: >-
+          header.match('header-filter', 'check-this-header-filter')
+        - key: example-overlay
+          expression: body.example.value
+    - webhook:
+        objectRef:
+          kind: Service
+          name: example-webhook-service
+          apiVersion: v1
+          namespace: tekton-pipelines
+  eventListeners:
```

Comment:
This would create services for things that we might not want to expose... on the plus side it would allow to reuse the trigger selection logic from the event listener.